### PR TITLE
Contact form: Limit the blocks that can be inserted to only core blocks and markdown

### PR DESCRIPTION
**Pull request by enejb**
This make sure that the user is only able to use some of the blocks. 

#### Changes proposed in this Pull Request
Adds a whitelist of blocks that a user can add to the form. 
This prevents cases where the user enter blocks that are currently broken and lead to a broken user flow. 

For example a user can right now insets a form into a form. 

Check the code for the list of supported blocks. 
I removed, button, columns, and the contact form. 

#### Testing instructions
Testing instructions

1. Check out this Jurassic Ninja link.
    https://jurassic.ninja/create/?gutenberg&gutenpack&shortlived&jetpack-beta&calypsobranch=add/contact-form-block-limit-available-blocks&branch=add/contact-form-gutenblock-sdk

2. Connect jetpack

3. Start a new page

4. Add a the new contact form block

5. Try adding all the fields. Can you see one that breaks everything on the frontend? That doesn't make sense?

### CODE REVIEW

**Comment by lezama** [comment]
columns could be useful, no?

File: client/gutenberg/extensions/contact-form/components/jetpack-contact-form.jsx [hunk 1]
```diff
@@ -11,6 +11,71 @@ import { Component, Fragment } from '@wordpress/element';
...
+	'core/pullquote',
+	'core/separator',
+	'core/block',
+	'core/spacer',
+	'core/subhead',
+	'core/table',
+	'core/template',
+	'core/verse',
+	'core/video',
+];
```
Comment (roccotripaldi) [inline client/gutenberg/extensions/contact-form/components/jetpack-contact-form.jsx]: Not a big fan of the whitelist approach. But it works as described!

A black list approach would look like:

```
import { getBlockTypes } from '@wordpress/blocks';

const DISALLOWED_BLOCKS = [
	'jetpack/contact-form'
];

const allowedBlocks = getBlockTypes().map( block => block.name ).filter( name => ! DISALLOWED_BLOCKS.includes( name ) );
```

**Comment by enejb** [comment]
> columns could be useful, no? 

Yes they would be. but they result in a broken experience right now :( 


@roccotripaldi I would rather limit the number of things to the things that work then allow the user to end up with something that looks broken. 

Lets go with a whitelist that works for now and then revisit this choice in V2.

File: client/gutenberg/extensions/contact-form/components/jetpack-contact-form.jsx (hunk 1 above)
Comment (mtias) [reply client/gutenberg/extensions/contact-form/components/jetpack-contact-form.jsx]: For so many items, whitelist doesn't make a lot of sense to me — I'd assume you want to keep it up to date with new blocks that come in core.

If the idea is to just add paragraph, image, and maybe one or two others, whitelist seems fine.

**Pull request merged**